### PR TITLE
Add filter backlinks option

### PR DESCRIPTION
**Pull request by Narigo**

### CODE REVIEW

File: src/filter.js [hunk 1]
```diff
@@ -111,6 +122,9 @@ function addDependenciesOfTable(allTables, accTables, currentTable, excludeBackl
       if (!ignoreMissing) {
         console.warn("Linking to a missing table - ignoring!", ignoreMissing);
       }
+    } else if (filterBacklinks && tableId === excludedTableId) {
+      // do not care about first table
+      console.log("back link to first table", tableId);
```
Comment (alexvetter) [inline src/filter.js]: Don't really know if term back link is correct. A link doesn't really have a direction. It is only a back link from the filters starting point. Also is this log necessary? Doesn't this pollute the aggregation log?

File: src/filter.spec.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ import filterFixture10 from "./__tests__/filterFixture10.json";
 import filterFixture11 from "./__tests__/filterFixture11.json";
 import filterFixture12 from "./__tests__/filterFixture12.json";
 import filterFixture13 from "./__tests__/filterFixture13.json";
+import filterFixtureA3AndA4 from "./__tests__/filterFixtureA3AndA4.json";
```
Comment (alexvetter) [inline src/filter.spec.js]: Folder structure and having tests next to source made it difficult for me to overlook the project and to see the project structure at a glance. Don't like it. But that's just a comment.

File: src/filter.spec.js [hunk 3]
```diff
@@ -379,4 +380,16 @@ describe("filter", () => {
...
+  it("can reuse the predicate on backlink-dependencies and filter them", () => {
+    return Promise
+      .resolve(cyclicTablesWithDependencyLinkFixture)
+      .then(filter({
+        filterBacklinks: true,
+        path: ["tableA"],
+        predicate: v => v.identifier === "A3" || v.identifier === "A4"
+      }))
+      .then(data => {
+        expect(data).to.eql(filterFixtureA3AndA4);
```
Comment (alexvetter) [inline src/filter.spec.js]: Should we add another test with row `id` and over three layers/tables?

File: src/filter.js [hunk 4]
```diff
@@ -40,10 +42,11 @@ export function filter(
           accTables[firstTable.id],
           excludeBacklinks,
           `${firstTable.id}`,
-          ignoreMissing
+          ignoreMissing,
+          filterBacklinks
         );
 
-        if (excludeBacklinks) {
+        if (excludeBacklinks || filterBacklinks) {
```
Comment (alexvetter) [inline src/filter.js]: That's were the magic happens?

File: src/filter.spec.js (hunk 3 above)
Comment (alexvetter) [reply src/filter.spec.js]: What about a test with two chained filters (e.g. modelYear <= 2018 and than salesRegion === USA)? Should be possible, right?

Comment (Narigo) [reply src/filter.spec.js]: There are tests for multiple layers links and also a test for multiple chained filters

File: src/filter.js (hunk 1 above)
Comment (Narigo) [reply src/filter.js]: Log will be removed

**Pull request merged**